be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/ElectionHandlerTest.scala

```diff
@@ -214,142 +218,144 @@ class ElectionHandlerTest extends TestKit(ActorSystem("Election-DB-System")) wit
     system.actorOf(dbActorMock)
   }
 
+  private def injectDb(dbRef: AskableActorRef) = PublishSubscribe.buildGraph(Actor.noSender, dbRef, Actor.noSender, MessageRegistry(), Actor.noSender, Actor.noSender, Actor.noSender, false)
```

Comment:
Can't we use something like: https://www.scalatest.org/scaladoc/1.8/org/scalatest/BeforeAndAfterEach.html ?

Reply:
in this case not, because different types of mocked db are injected. Using BeforeAndAfterEach would force to inject the same db for all test.